### ISSUE TITLE
shush_backtraces fails due to SystemExit#backtrace → nil (Ruby 2.5.0)

### ISSUE THREAD

**Issue by mattbrictson** [rails/spring#550]
Ruby 2.4.3:

```
$ bin/rails routes
Running via Spring preloader in process 56672
     Prefix Verb URI Pattern Controller#Action
sidekiq_web      /sidekiq    Sidekiq::Web
       root GET  /           home#index
```

Ruby 2.5.0:

```
$ bin/rails routes
Running via Spring preloader in process 59563
     Prefix Verb URI Pattern Controller#Action
sidekiq_web      /sidekiq    Sidekiq::Web
       root GET  /           home#index
rake aborted!
NoMethodError: undefined method `reject!' for nil:NilClass
Traceback (most recent call last):
/Users/mbrictson/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/application.rb:171:in `fork': undefined method `reject!' for nil:NilClass (NoMethodError)
```

I traced the cause to Spring's `shush_backtraces`, specifically this line:

https://github.com/rails/spring/blob/b7aaf1752212769a9f04a51282e6500ef3db7959/lib/spring/application.rb#L291

It seems that in Ruby 2.5.0, `$!.backtrace` is `nil` for some cases of the `SystemExit` exception. This causes `shush_backtraces` to crash.

**Comment by mattbrictson** [comment]
Suggested fix: #551

**Comment by mattbrictson** [comment]
This issue has been fixed by https://github.com/banister/binding_of_caller/pull/69. Upgrading to `binding_of_caller >= 0.8.0` resolves the problem.
